### PR TITLE
AMB voltage and current models with CAN TX

### PR DESCRIPTION
**Pull request by ishanjoshi23**
Current AMB FW Updates: Voltage and Current models implemented; still waiting on Thermistors to implement temperature models. 
CAN support is also added.

### CODE REVIEW

File: components/mcb/mcb_main/.project
Comment (MatthewChow03) [inline components/mcb/mcb_main/.project:1]: Not sure why this is here. Probably should remove.

File: .project
Comment (MatthewChow03) [inline .project:1]: Remove

**Review by MatthewChow03 — DISMISSED** [review]
Lgtm, pre-emptive approval. You might want to add a `.gitignore` inside `components/amb`.

File: .project
Comment (ishanjoshi23) [reply .project:1]: Sure, will remove it

File: components/mcb/mcb_main/.project
Comment (ishanjoshi23) [reply components/mcb/mcb_main/.project:1]: Removed it

**Comment by a2k-hanlon** [comment]
The AMB project is a CubeIDE project wit a `.project` file of its own, so I don't think ignoring all .project files is a good idea. Removing the stray mcb .project file should be suffice.

**Comment by MatthewChow03** [comment]
@ishanjoshi23 I meant something like the project specific .gitignore like https://github.com/UBC-Solar/firmware_v3/blob/master/components/mcb/mcb_main/.gitignore

But I didn't realize you have things set up. For example, you have a git ignore inside the Debug folder. If the amb setup ignores all the random build files then it should be fine

**Comment by a2k-hanlon** [comment]
@MatthewChow03 Ishan did make a project specific gitignore `components/amb/.gitignore`. But, it's contents are just `*.project`. I wouldn't recommend that because it ignores the amb project's `.project` file.

If you want to go for consistency with other projects, keep `components/amb/.gitignore`, remove `components/amb/Debug/.gitignore`, and put something like this in `components/amb/.gitignore`:

```
# Build outputs
Debug/
Release/

# Project log
.mxproject

# Debug configuration
*.cfg
*.launch

```

**Comment by ishanjoshi23** [comment]
> @MatthewChow03 Ishan did make a project specific gitignore `components/amb/.gitignore`. But, it's contents are just `*.project`. I wouldn't recommend that because it ignores the amb project's `.project` file.
> 
> If you want to go for consistency with other projects, keep `components/amb/.gitignore`, remove `components/amb/Debug/.gitignore`, and put something like this in `components/amb/.gitignore`:
> 
> ```
> # Build outputs
> Debug/
> Release/
> 
> # Project log
> .mxproject
> 
> # Debug configuration
> *.cfg
> *.launch
> ```

**Comment by ishanjoshi23** [comment]
@MatthewChow03 @a2k-hanlon I've removed the .gitignore from /Debug/ and included the above mentioned contents in the gitignore in /components/amb/.gitignore

**Comment by a2k-hanlon** [comment]
Looks good! You may want to also remove, and commit the removal of, the files that are now ignored including the `.launch` file and the contents of the `Debug` folder

**Comment by ishanjoshi23** [comment]
Ok, I have removed those.

**Review by MatthewChow03 — APPROVED** [review]

**Pull request merged**